### ISSUE TITLE
Object3D's destroy() removes from the scene all the lights bound to the object's material

### ISSUE THREAD

**Issue by manjia** [Rajawali/Rajawali#1718]
#### Rajawali Version or Branch
master

#### Device and Android Version
Found on a Samsung Galaxy S6 SM-G925F and Samsung Galaxy Tab 4 SM-T535, but I suspect it is device independent.

#### Summary
Object3D's destroy() removes from the scene all the lights binded to the object's material

#### Steps to Reproduce
Add a light to the scene:

```
    DirectionalLight directionalLight = new DirectionalLight(1f, .2f, -1.0f);
    directionalLight.setColor(1.0f, 1.0f, 1.0f);
    directionalLight.setPower(2);
    getCurrentScene().addLight(directionalLight);
```

add an object to the scene:

```
    Object3D cube = new Cube(0.04f);
    Material material = new Material();
    material.setColor(Color.GREEN);
    material.setDiffuseMethod(new DiffuseMethod.Lambert());
    material.enableLighting(true);
    cube.setMaterial(material);
    getCurrentScene().addChild(cube);
```

remove and destroy the object:

```
    getCurrentScene().removeChild(cube);
    cube.destroy();
```

add another object to the scene:

```
    Object3D cube2 = new Cube(0.04f);
    Material material2 = new Material();
    material2.setColor(Color.RED);
    material2.setDiffuseMethod(new DiffuseMethod.Lambert());
    material2.enableLighting(true);
    cube2.setMaterial(material2);
    getCurrentScene().addChild(cube2);

    // I also invalidate the lights, just to be sure
    getCurrentScene().markLightingDirty();
```

This ends up in having the red cube with no lights applied to its material, by doing some debug I confirmed that it happens because when I add the 2nd cube the scene has got no lights (mLights.size() is zero) even tough I didn't remove the directionalLight that was previously added.

By doing further debug I noticed that calling Object3D's destroy() method triggers
```
MaterialManager.getInstance().removeMaterial(mMaterial); 
```
in Object3D class, which consequently triggers `removeMaterial` on Renderer class which consequently triggers `taskRemove` on MaterialManager class which consequently triggers `remove` on Material class which clears the Material's mLights list, this list points to the same mLights list object belonging to the scene itself (it gets set there on scene render, the method chain goes pretty much as follows: Scene's `render` -> Scene's `updateMaterialsWithLights()` -> Scene's `updateChildMaterialWithLights(Object3D child)` -> Material's `setLights(List<ALight> lights)` -> `mLights = lights`), so clearing the Material's mLights ends up in also clearing the scene's lights

I would like to help fixing this problem but I'm not so sure how the lights should be linked to the Material to work properly (copy? weak reference?), if you can address me to the right solution I'll give it a try. 

Thanks

**Comment by contriteobserver** [comment]
reproduced with the following code, however, if one changes the order, so the new cube is created before the old one is destroyed (just swap the timing delays), the lights remain.
```
        @Override
        protected void initScene() {
            try {
                getCurrentScene().setBackgroundColor(Color.BLUE);;

                DirectionalLight key = new DirectionalLight(2,2,0);
                key.setPosition(-2,-2,0);
                key.setPower(4);
                getCurrentScene().addLight(key);

                Material material = new Material();
                material.setColor(Color.GREEN);
                material.setDiffuseMethod(new DiffuseMethod.Lambert());
                material.enableLighting(true);
                cube = new Cube(1);
                cube.setMaterial(material);
                getCurrentScene().addChild(cube);

                getCurrentCamera().setPosition(3,3,3);
                getCurrentCamera().setLookAt(cube.getPosition());

                new Timer().schedule(new TimerTask() {
                    @Override
                    public void run() {
                        // this code will be executed after 2 seconds
                        getCurrentScene().removeChild(cube);
                        cube.destroy();
                    }
                }, 2000);

                new Timer().schedule(new TimerTask() {
                    @Override
                    public void run() {
                        // this code will be executed after 4 seconds
                        Object3D cube2 = new Cube(1);
                        Material material2 = new Material();
                        material2.setColor(Color.RED);
                        material2.setDiffuseMethod(new DiffuseMethod.Lambert());
                        material2.enableLighting(true);
                        cube2.setMaterial(material2);
                        getCurrentScene().addChild(cube2);
                    }
                }, 4000);
            } catch (Exception e) {
                Log.d(getLocalClassName(), e.toString());
            }
        }
```

**Comment by manjia** [comment]
@contriteobserver Good point, expecially considering that in a regular scenario many Object3Ds will be loaded on scene initialization and I see no point in removing all the objects from the scene, meaning that at least one object will always be present and thus the scene lights will be preserved.

In fact I came across this issue in a situation where I needed to load several objects one by one in subsequent steps leaving the scene empty in the meanwhile (because my 3d models are very heavy and loading all of them together leads to an OutOfMemory error).

Even though the scenario I described happens rarely (and in that case it can be easily fixed ) I'm wondering if it is worth to fix this issue anyway or not, I mean, is it correct to remove all the scene lights when there's no more object to show into the scene? In my opinion it could even be a correct behavior, I'm just asking because I'm not that much familiar with 3d cgi best practices, if this behavior is considered correct or acceptable to me this issue can be closed right now, if it is not I think it's worth to fix it.
